applications/serial_lte_modem/src/main.c

```diff
@@ -333,6 +336,43 @@ static void handle_mcuboot_swap_ret(void)
 	}
 }
 
+int nidd_auto_connect(void)
+{
+	int err = 0;
+
+#if defined(CONFIG_LWM2M_CARRIER_SERVER_BINDING_N)
+	int stat;
+
+	if (!auto_connect) {
+		return 0;
+	}
+	err = nrf_modem_at_scanf("AT+CEREG?", "+CEREG: %d", &stat);
+	if (err != 1 || (stat == 1 || stat == 5)) {
+		return 0;
+	}
+
+	LOG_INF("NIDD auto connect");
+	err = nrf_modem_at_printf("AT%%XSYSTEMMODE=0,1,0,0");
+	if (err) {
+		LOG_ERR("Failed to configure RAT: %d", err);
+		return err;
+	}
+	/* NOTE - PDP Context for NIDD needs to be configured per network */
+	err = nrf_modem_at_printf("AT+CGDCONT=0,\"Non-ip\"");
+	if (err) {
+		LOG_ERR("Failed to define NIDD PDN: %d", err);
+		return err;
+	}
+	err = nrf_modem_at_printf("AT+CFUN=1");
```

Comment:
As per my suggestion, I think CFUN=1 does not need to be restricted to NIDD binding, but maybe just CONFIG_SLM_CARRIER.